### PR TITLE
feat(api): converted bulk trigger to use SDK

### PR DESCRIPTION
**Pull request by tatarco**
### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/api/src/app/events/e2e/bulk-trigger.e2e.ts [hunk 1]
```diff
@@ -1,13 +1,18 @@
 import { expect } from 'chai';
-import axios from 'axios';
 import { MessageRepository, NotificationRepository, NotificationTemplateEntity, SubscriberEntity } from '@novu/dal';
-import { UserSession, SubscribersService } from '@novu/testing';
+import { SubscribersService, UserSession } from '@novu/testing';
 import { ChannelTypeEnum, StepTypeEnum } from '@novu/shared';
-
-const axiosInstance = axios.create();
+import { Novu } from '@novu/api';
+// eslint-disable-next-line no-restricted-imports
```
Comment (SokratisVidros) [inline apps/api/src/app/events/e2e/bulk-trigger.e2e.ts]: Is this eslint directive required? If not, let's remove.

File: apps/api/src/app/shared/helpers/e2e/sdk/e2e-sdk.helper.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import { UserSession } from '@novu/testing';
+// eslint-disable-next-line no-restricted-imports
+import { SDKError } from '@novu/api/models/errors/sdkerror';
+import { expect } from 'chai';
+
+export function initNovuClassSdk(session: UserSession): Novu {
+  return new Novu({ apiKey: session.apiKey, serverURL: session.serverUrl, debugLogger: console });
+}
+export function initNovuFunctionSdk(session: UserSession): NovuCore {
+  return new NovuCore({ apiKey: session.apiKey, serverURL: session.serverUrl, debugLogger: console });
```
Comment (SokratisVidros) [inline apps/api/src/app/shared/helpers/e2e/sdk/e2e-sdk.helper.ts:11]: What is the difference between Novu and NovuCore in the generated SDK?

File: apps/api/src/app/events/e2e/bulk-trigger.e2e.ts [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
+import { NovuCore } from '@novu/api/core';
```
Comment (SokratisVidros) [inline apps/api/src/app/events/e2e/bulk-trigger.e2e.ts]:
```suggestion
import type { NovuCore } from '@novu/api/core';
```

File: apps/api/src/app/events/e2e/bulk-trigger.e2e.ts [hunk 4]
```diff
@@ -1,13 +1,18 @@
...
+// eslint-disable-next-line no-restricted-imports
+import { triggerBulk } from '@novu/api/funcs/triggerBulk';
```
Comment (SokratisVidros) [inline apps/api/src/app/events/e2e/bulk-trigger.e2e.ts:9]:
```suggestion
import type { triggerBulk } from '@novu/api/funcs/triggerBulk';
```

File: apps/api/src/app/events/e2e/bulk-trigger.e2e.ts [hunk 5]
```diff
@@ -212,73 +211,67 @@ describe('Trigger bulk events - /v1/events/trigger/bulk (POST)', function () {
...
+      ],
+    });
+    if (!bulkTriggerResponse.ok) {
+      throw new Error(`failed to bulk trigger:${JSON.stringify(bulkTriggerResponse.error)}`);
+    }
 
-    const { data: body } = response;
-    expect(body.data).to.be.ok;
-    expect(body.data.length).to.equal(3);
+    const dtoList = bulkTriggerResponse.value.result;
```
Comment (SokratisVidros) [inline apps/api/src/app/events/e2e/bulk-trigger.e2e.ts:260]: This `.value.result` duplication is subpar DX. Can we remove the `.value` part?

File: apps/api/src/app/notifications/notification.controller.ts [hunk 6]
```diff
@@ -65,20 +65,21 @@ export class NotificationsController {
...
+      environmentId: user.environmentId,
+      userId: user._id,
+      channels: channelsQuery,
+      templates: templatesQuery,
+      emails: emailsQuery,
+      search: query.search,
+      subscriberIds: subscribersQuery,
+      transactionId: query.transactionId,
+    });
+    console.log('command', command);
```
Comment (SokratisVidros) [inline apps/api/src/app/notifications/notification.controller.ts]: Please remove and revert this change to its previous, more compact version.

**Review by SokratisVidros — CHANGES_REQUESTED** [review]

File: apps/api/src/app/events/e2e/bulk-trigger.e2e.ts (hunk 4 above)
Comment (tatarco) [inline apps/api/src/app/events/e2e/bulk-trigger.e2e.ts:6]: cannot do that, it's failing compilation, if importing type the method cannot be called, it's for typescript compilation and intelisense only

File: apps/api/src/app/events/e2e/bulk-trigger.e2e.ts (hunk 5 above)
Comment (tatarco) [reply apps/api/src/app/events/e2e/bulk-trigger.e2e.ts:260]: the value belongs to the reilroad programming, the result belongs to the response always being wrapped, we get the body and the headers where the body is in the results, what do you want to do?

File: apps/api/src/app/shared/helpers/e2e/sdk/e2e-sdk.helper.ts (hunk 2 above)
Comment (tatarco) [reply apps/api/src/app/shared/helpers/e2e/sdk/e2e-sdk.helper.ts:11]: Novu is the Client and the base line for all of the regular api's so it's (new Novu()).organizations.create() 
novu core is the first parameter in createOrganization(new NovuCore(), dto)

File: apps/api/src/app/events/e2e/bulk-trigger.e2e.ts (hunk 5 above)
Comment (SokratisVidros) [reply apps/api/src/app/events/e2e/bulk-trigger.e2e.ts:260]: Ideally, it should be `bulkTriggerResponse.result` and `bulkTriggerResponse.error`, right? Do we have leverage to change that on the speakeasy generated code?

**Review by SokratisVidros — APPROVED** [review]

File: apps/api/src/app/events/e2e/bulk-trigger.e2e.ts (hunk 5 above)
Comment (SokratisVidros) [reply apps/api/src/app/events/e2e/bulk-trigger.e2e.ts:260]: This was resolved offline. We will chat with speakeasy to get flatter responses.

File: apps/api/src/app/shared/helpers/e2e/sdk/e2e-sdk.helper.ts (hunk 2 above)
Comment (SokratisVidros) [reply apps/api/src/app/shared/helpers/e2e/sdk/e2e-sdk.helper.ts:11]: This was resolved offline. We will chat with the Speakeasy.

**Review by SokratisVidros — APPROVED** [review]
